sphinx/testing/util.py

```diff
@@ -204,14 +207,65 @@ def build(self, force_all: bool = False, filenames: list[str] | None = None) ->
         super().build(force_all, filenames)
 
 
+class SphinxTestAppLazyBuild(SphinxTestApp):
```

Comment:
This seems logically similar to `SphinxTestAppWrapperForSkipBuilding` - is there a reason we can't use that instead?

Reply:
There are real differences. First is that the other class is a proxy class so you don't have a direct inheritance. The second is that it does not check the usage of the class and I don't want to make tests using the old plugin fail because of that.

Reply:
Should it be added to `__all__`?  (I don't use that feature often but I do notice that `SphinxTestAppWrapperForSkipBuilding` is in there)

Reply:
I personally think that this class is an implementation detail and not part of the public scope but I don't want to put an `_` beforehand since it could mean private for the module itself (but it's only private for outside of Sphinx itself). So I will not export the new ones (though I will leave those that are already exported in the public scope).

Reply:
Sounds good.  Thank you :)